buildSrc/src/main/java/com/uber/okbuck/core/dependency/DependencyUtils.java

```diff
@@ -122,17 +122,14 @@ static Path getSingleZipFilePath(Project project, File baseDir, String zipToFind
             ImmutableMap.of(
                 "dir", baseDir.getAbsolutePath(), "includes", ImmutableList.of("**/" + zipToFind)));
 
-    try {
-      File maybeZipFile = zipFiles.getSingleFile();
-      if (FileUtil.isZipFile(maybeZipFile)) {
-        return maybeZipFile.toPath();
-      }
-    } catch (IllegalStateException ignored) {
-      if (zipFiles.getFiles().size() > 1) {
-        throw new IllegalStateException("Found multiple source jars: " + zipFiles);
-      }
-    }
-    return null;
+    return zipFiles
+        .getFiles()
+        .stream()
+        .sorted()
+        .filter(FileUtil::isZipFile)
+        .findFirst()
+        .map(File::toPath)
+        .orElse(null);
```

Comment:
is it possible to maybe get the most recent one?

also, we can probably print a warning for the user that you might see cache misses and should clear these folders to prevent it in future?

Reply:
I would recommend we fix this by making a custom version of the bad artifacts instead of allowing non determinism in the builds

Another option is to make buck not add the source jar to the rulekey

Reply:
we already have this non-determinism depending on which of the source jars given machine fetched. The only thing that we were getting on top is we were getting random failures sometimes in some cases blocking builds and needing immediate attention to proceed.